feat(MA): adding colors and perc to cells in MA table

jabahamondes opened the pull request:
## Changes

Many changes. This PR includes:
- adding percentage instead of previous value making a faster understanding of the performance
- refactoring the response to support previous, current, perc and isIncreaseBad (as web overview does)
- adding colors depending on the value and isIncreaseBad
- renaming columns to use less space into cost, clicks, impressions, cpc
- handling diverse case of scenarios for the table: when there are 0s or null or strings involved

Note: it's tricky when handling 0 and null values in current vs previous states (null would be for example when there is 0 conversion so cost per conversion would be null)

## How did you test this code?
Updated test and manually:
<img width="675" height="490" alt="image" src="https://github.com/user-attachments/assets/58ef5e08-ed0d-4089-93a9-1a97a1efd8c8" />

<img width="334" height="89" alt="image" src="https://github.com/user-attachments/assets/2e26e37b-ef78-4ca7-a03c-be65059128b9" />

<img width="354" height="80" alt="image" src="https://github.com/user-attachments/assets/4a303304-7fcd-4ee2-b304-989d79dcf0e8" />

<img width="382" height="77" alt="image" src="https://github.com/user-attachments/assets/07126197-f51f-4dc5-b5b0-23ff7d1201d6" />

<img width="338" height="86" alt="image" src="https://github.com/user-attachments/assets/2cfc1483-dba8-452a-9e92-e5778335239d" />

<img width="436" height="120" alt="image" src="https://github.com/user-attachments/assets/61b819b1-581f-40ab-ad0c-4bfa33f07003" />

<img width="750" height="38" alt="image" src="https://github.com/user-attachments/assets/0a615d98-8626-4521-bf4b-49a20db797f2" />

Commit:
feat(analytics): update schema and types for marketing analytics
- Adding perc and colors to cells
- Added  definition with properties for change percentage, comparison, and value types.
- Updated  to replace  references in various files.
- Modified  to use concise naming for cost-related fields.
- Introduced  enum for handling extreme values in analytics calculations.
- Enhanced formatting functions to accommodate new item types and handle infinity cases in percentage changes.

This update improves the structure and clarity of marketing analytics data handling across the application.

github-actions[bot] committed:
Update query snapshots

Commit:
feat(analytics): enhance marketing analytics data handling
- Added type guard for valid numeric comparison data in shared.tsx.
- Updated tooltip functions to ensure valid data is used for percentage changes.
- Refactored transformation logic in marketing_analytics_table_query_runner.py to streamline result processing into MarketingAnalyticsItem objects.
- Adjusted test cases to ensure proper type handling for campaign performance ranking.

These changes improve data integrity and clarity in marketing analytics calculations and visualizations.

Commit:
Merge branch 'feat/adding-colors-and-perc-to-table-comparison' of github.com:PostHog/posthog into feat/adding-colors-and-perc-to-table-comparison

Commit:
Merge remote-tracking branch 'origin/master' into feat/adding-colors-and-perc-to-table-comparison

lricoy commented on products/marketing_analytics/backend/hogql_queries/constants.py:
```diff
@@ -145,3 +154,130 @@
+        and isinstance(value, int | float)
+        and isinstance(previous, int | float)
+    ):
+        try:
+            if previous == 0:
+                # Handle special cases when previous is 0
+                if value == 0:
+                    # Both are 0: no change
+                    change_from_previous_pct = 0
+                elif value > 0:
+                    # From 0 to positive: use special large number to represent infinite growth
+                    change_from_previous_pct = int(InfinityValue.NUMBER_999999)
+                else:
+                    # From 0 to negative: use special large negative number to represent infinite decrease
+                    change_from_previous_pct = int(InfinityValue.NUMBER__999999)
+            else:
+                # Normal case: previous != 0
+                change_from_previous_pct = round(100 * (value - previous) / previous)
+        except (ValueError, ZeroDivisionError):
+            pass
```
Same as the other PR, I don't think we do those 'Infinity' differences, just consider the actual general improvement.

lricoy reviewed (approved):
LGTM

I think the background color may be too much, I would just change the font color at first. 

If you do decide to keep it, I would remove the padding and make the table fill the whole thing so there is not white bg between the cells.

Commit:
merge conflicts

Commit:
fix(analytics): Update MarketingAnalyticsCell and styles for better rendering

- Changed the return value for infinite percentage cases to '-' for consistency.
- Enhanced  to accept an optional  prop for better customization.
- Updated rendering logic in  to apply maxWidth conditionally based on column name.
- Added CSS rules to ensure table cells and rows utilize full height for improved layout.

Commit:
chore: Refactor imports and clean up test files

- Removed unused import from .
- Reordered imports in  for consistency.
- Added missing import back to .
- Cleaned up whitespace in .
